lib/devise/multi_email/models/authenticatable.rb

```diff
@@ -48,22 +48,32 @@ def email=(new_email)
       module ClassMethods
         def find_first_by_auth_conditions(tainted_conditions, opts = {})
           filtered_conditions = devise_parameter_filter.filter(tainted_conditions.dup)
-          email = filtered_conditions.delete(:email)
+          criteria = filtered_conditions.extract!(:email, :unconfirmed_email)
 
-          if email && email.is_a?(String)
+          if criteria.keys.any?
             conditions = filtered_conditions.to_h.merge(opts).
-              reverse_merge(multi_email_association.reflection.table_name => { email: email })
+              reverse_merge(build_conditions criteria)
 
             resource = joins(multi_email_association.name).find_by(conditions)
-            resource.current_login_email = email if resource.respond_to?(:current_login_email=)
+            resource.current_login_email = criteria.values.first if resource
```

Comment:
Sure, you can see the attribute is created here: https://github.com/identification-io/devise-multi_email/blob/feature/progressive_refactor/lib/devise/multi_email/models/authenticatable.rb#L17

That's when the module is included in the parent/user model, and then the finder method is a class method on that same model.